apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/grpc/StargateBridgeInterceptor.java

```diff
@@ -0,0 +1,34 @@
+package io.stargate.sgv2.api.common.grpc;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.quarkus.grpc.GlobalInterceptor;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@GlobalInterceptor
+@ApplicationScoped
+public class StargateBridgeInterceptor implements ClientInterceptor {
+  @Inject StargateRequestInfo requestInfo;
+  @Inject GrpcClients grpcClients;
```

Comment:
Propose to inject `StargateRequestInfo` and `GrpcMetadataConfig config` in the constructor as in the `GrpcClients`.. The construct the  keys:

```
  /** Metadata key for passing the tenant-id to the Bridge. */
  private final Metadata.Key<String> tenantIdKey;
  /** Metadata key for passing the cassandra token to the Bridge. */
  private final Metadata.Key<String> cassandraTokenKey;
```

And use them to construct the metadata passed to the `HeaderAttachingClientCall`.